src/algorithms/model.ts

```diff
@@ -0,0 +1,376 @@
+import { AllParamsFlat } from '../algorithms/types/Param.types'
+import { SeverityTableRow } from '../components/Main/Scenario/SeverityTable'
+import { ModelParams, SimulationTimePoint } from '../algorithms/types/Result.types'
+
+const msPerDay = 1000 * 60 * 60 * 24
+
+const monthToDay = (m: number) => {
+  return m * 30 + 15
+}
+
+const jan2020 = new Date('2020-01-01').valueOf() // time in ms
+
+/**
+ *
+ * @param time - point in time, until which simulation runs, as epoch time
+ * @param avgInfectionRate
+ * @param peakMonth - counting number in range of 0-11
+ * @param seasonalForcing -  seasonal variation in transmission. Usually a decimal number, e. g. 0.2
+ * @returns
+ */
+export function infectionRate(
+  time: number,
+  avgInfectionRate: number,
+  peakMonth: number,
+  seasonalForcing: number,
+): number {
+  // this is super hacky
+  const phase = ((time - jan2020) / msPerDay / 365 - monthToDay(peakMonth) / 365) * 2 * Math.PI
+  return avgInfectionRate * (1 + seasonalForcing * Math.cos(phase))
+}
+
+export function getPopulationParams(
+  params: AllParamsFlat,
+  severity: SeverityTableRow[],
+  ageCounts: Record<string, number>,
+  containment: (t: Date) => number,
+): ModelParams {
+  const timeDeltaDays = 0.25
+
+  // TODO: Make this a form-adjustable factor
+  const pop: ModelParams = {
+    ageDistribution: {},
+    infectionSeverityRatio: {},
+    infectionFatality: {},
+    infectionCritical: {},
+    hospitalizedRate: {},
+    recoveryRate: {},
+    dischargeRate: {},
+    stabilizationRate: {},
+    criticalRate: {},
+    deathRate: {},
+    overflowDeathRate: {},
+    isolatedFrac: {},
+    importsPerDay: { total: params.importsPerDay },
+    timeDeltaDays,
+    // Dummy infectionRate function. This is set below.
+    infectionRate: () => -Infinity,
+    timeDelta: msPerDay * timeDeltaDays,
+    incubationTime: params.incubationTime,
+    populationServed: params.populationServed,
+    numberStochasticRuns: params.numberStochasticRuns,
+    hospitalBeds: params.hospitalBeds,
+    ICUBeds: params.ICUBeds,
+  }
```

Comment:
The initialization is now occurring in a single line rather than adding fields one my one.